include/TrustWalletCore/TWCurve.h

```diff
@@ -12,9 +12,10 @@ TW_EXTERN_C_BEGIN
 /// Elliptic cruves
 TW_EXPORT_ENUM()
 enum TWCurve {
-    TWCurveSECP256k1       /* "secp256k1" */,
-    TWCurveEd25519         /* "ed25519" */,
-    TWCurveNIST256p1       /* "nist256p1" */,
+    TWCurveSECP256k1              /* "secp256k1" */,
+    TWCurveEd25519                /* "ed25519" */,
+    TWCurveEd25519Blake2bNano     /* "ed25519-blake2b-nano" */,
```

Comment:
This should only represent elliptic curves, not hash functions. I know this deviates from the tremor-crypto implementation, but I think it's clearer to have that separation.

Reply:
The TWCurve also controls the bip32-seed string used in HDWallet. And later the key derivation algorithm selected (eg ed25519-blake2b, instead of ed25519-sha512). How would you propose to approach this?

Reply:
I think each of those things should be separate. You need to specify both a curve and a key derivation algorithm.

Reply:
Quite a lot of APIs would have to be refactored then to accept TWCurve and TWNewEnum.